docs/architecture/high-level-components.md

```diff
@@ -0,0 +1,29 @@
+# High Level Components
```

Comment:
Please add an entry (as a first one) in /docs/README.md inside architecture section